data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/export/ExportPartitionWorker.java

```diff
@@ -177,6 +186,9 @@ public void run() {
                             LOG.debug("Write to buffer for line " + (recordCount - DEFAULT_BATCH_SIZE) + " to " + recordCount);
```

Comment:
I have seen `(recordCount - DEFAULT_BATCH_SIZE)` value reporting negative. Need to fix this.

Reply:
`recordCount` seems to be a duplicate of `totalRecords` and is always greater than `startLine` at this point. Is it possible that recordCount (an int) overflows, causing the negative number? `recordCount` should probably be of long type.

Reply:
The export partition size is an int, so recordCount shouldn't overflow. I can't see how `recordCount - DEFAULT_BATCH_SIZE` becomes negative.